### PR TITLE
feat: add basic support openApi 3.1

### PR DESCRIPTION
**Pull request by AlexVarchuk**
fixes https://github.com/Redocly/openapi-cli/issues/276

### CODE REVIEW

**Review by andriyl — APPROVED** [review]

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:17]: No need for x-webhooks

Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:17]: No need for x-webhooks in 3.1
```suggestion
```

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:15]:
```suggestion
    webhooks: 'WebhooksMap',
```

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:22]: can we reuse shared types from oas3.0 ?

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:130]:
```suggestion
```

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:376]: formatting

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:352]: you should add more fields here, new json schema draft supports `if`, `else`, `const`, etc.

File: packages/core/src/types/oas3_1.ts
Comment (RomanHotsiy) [inline packages/core/src/types/oas3_1.ts:510]: There is one more added I think, `mutualTls`

**Review by RomanHotsiy — CHANGES_REQUESTED** [review]
Left a bunch of comments.

**Review by Marshevskyy — COMMENTED** [review]
good job Alex!

it would be amazing to add some tests which can cover this functionality,
we do not need to cover some implementation, 
but we should cover behavior

ideally we should somehow pass as an input OAS 3.1 and got it valid in linter....

**Comment by AlexVarchuk** [comment]
> good job Alex!
> 
> it would be amazing to add some tests which can cover this functionality,
> we do not need to cover some implementation,
> but we should cover behavior
> 
> ideally we should somehow pass as an input OAS 3.1 and got it valid in linter....

Good idea. I will try to implement it.

**Comment by AlexVarchuk** [comment]
@RomanHotsiy @Marshevskyy Please check it again. Thanks.

**Review by Marshevskyy — APPROVED** [review]
Alex! well done! 
tests 💪🏼 

but before merge, I would like to get approval from @RomanHotsiy

**Comment by timbunce** [comment]
Just a note from a random bystander to say Thanks Alex! Looking forward to using this.

**Review by RomanHotsiy — APPROVED** [review]
Good job 👏

**Pull request merged**
